components/automate-deployment/pkg/target/local_target.go

```diff
@@ -974,7 +974,34 @@ func (t *LocalTarget) DestroyPkgCache() error {
 	ctxLog := logrus.WithFields(logrus.Fields{
 		"path": habPath,
 	})
-	err := os.RemoveAll(habPath)
+
+	// If /hab is a mount point, remove the contents only.
+	mounts, err := ioutil.ReadFile("/proc/mounts")
```

Comment:
If interested, we also have

https://github.com/chef/automate/blob/master/lib/proc/procparse.go#L71

which will return you a list of the mount point, but the existing code is also very clear, so probably not worth changing it.